test/object-store/sync/flx_role_change.cpp

```diff
@@ -0,0 +1,860 @@
+                            std::optional<std::string_view> expected = std::nullopt) {
+            REQUIRE(!wait_for_download(*realm));
+            REQUIRE(!wait_for_upload(*realm));
+            wait_for_advance(*realm);
+            // Verify none of the records modified above exist in the realm
+            auto table = realm->read_group().get_table("class_Person");
+            REQUIRE(table->size() == cnt);
+            auto id_col = table->get_column_key("_id");
+            auto role_col = table->get_column_key("role");
+            for (auto& id : saved_ids) {
+                auto objkey = table->find_first(id_col, id);
+                if (expected) {
+                    REQUIRE(objkey);
+                    auto obj = table->get_object(objkey);
+                    REQUIRE(obj.get<String>(role_col) == *expected);
+                }
+                else {
+                    REQUIRE(!objkey);
+                }
+            }
+        };
+        // Update the rules so employees are not allowed and removed from view
+        // This will also remove the existing changes to the 10 employee records
+        // and the 5 new employee records.
+        size_t num_to_create = 5;
+        // Update 10 employees to worker-bee and create 5 new employees
+        update_records(test_realm, "employee", emp_ids, 10, num_to_create);
+        // Update 5 managers to worker-bee and create 5 new managers
+        update_records(test_realm, "manager", mgr_ids, 5, num_to_create);
+        // Update the allowed roles to "manager" and "worker-bee"
+        do_update_rules({{"role", {{"$in", {"manager", "worker-bee"}}}}});
+        // Resume the session and verify none of the new/modified employee
+        // records are present
+        test_realm->sync_session()->resume();
+        // Verify none of the employee object IDs are present in the local data
+        do_verify(test_realm, params.num_mgrs + num_to_create, emp_ids, std::nullopt);
+        // Verify all of the manager object IDs are present in the local data
+        do_verify(test_realm, params.num_mgrs + num_to_create, mgr_ids, "worker-bee");
+
+        // Update the allowed roles to "employee"
+        do_update_rules({{"role", "employee"}});
+        // Verify the items with the object IDs are still listed as employees
+        do_verify(test_realm, params.num_emps, emp_ids, "employee");
+
+        // Tear down the app since some of the records were added and modified
+        harness.reset();
+    }
+}
+
+TEST_CASE("flx: role changes during bootstrap complete successfully", "[sync][flx][baas][role change][bootstrap]") {
+    auto logger = util::Logger::get_default_logger();
+    static std::unique_ptr<FLXSyncTestHarness> harness;
+
+    // 150 emps, 25 mgrs, 10 dirs
+    // 10 objects before flush
+    // 1536 download soft max bytes
+    TestParams params{};
+    params.max_download_bytes = 1536;
```

Comment:
what's the meaning of this value?

Reply:
it is sent to the server to configure the "`qbs_download_changeset_soft_max_byte_size`" parameter that limits the number of bytes of changeset data in a download message. This is used to break up the bootstrap into multiple messages to test the multi-part bootstrap download and to "slow down" the bootstrap processing a bit.

Reply:
I know that, but why this specific value?

Reply:
It is totally arbitrary - I wanted a value more than 1K and less than 2K, so I chose 1.5 KB (1024 * 1.5) = 1536